src/rbprm-rom-validation.cc

```diff
@@ -80,8 +81,8 @@ namespace hpp {
           }
 
           if(!first){
-            collisionPairs_.erase(it);
-            collisionPairs_.insert(collisionPairs_.begin(),colPair);
+            pairs().erase(it);
+            pairs().insert(pairs().begin(),colPair);
```

Comment:
I propose to replace this code so that the collision requests are also reordered:

          std::size_t i = 0;
          for(; i < cPairs_.size() ; ++i)
            if(cPairs_[i].second == obj2)
              break;

          if(i == 0 && !cPairs_.empty()) {
            std::swap(cPairs_   [0], cPairs_   [i]);
            std::swap(cRequests_[0], cRequests_[i]);
          }

Reply:
You mean` i != 0` ?

Reply:
You are right.

Reply:
Done in eab7b4a.